lite/examples/object_detection/ios/ObjectDetection/ModelDataHandler/ModelDataHandler.swift

```diff
@@ -268,28 +269,56 @@ class ModelDataHandler: NSObject {
     isModelQuantized: Bool
   ) -> Data? {
     CVPixelBufferLockBaseAddress(buffer, .readOnly)
-    defer { CVPixelBufferUnlockBaseAddress(buffer, .readOnly) }
-    guard let mutableRawPointer = CVPixelBufferGetBaseAddress(buffer) else {
+    defer {
+      CVPixelBufferUnlockBaseAddress(buffer, .readOnly)
+    }
+    guard let sourceData = CVPixelBufferGetBaseAddress(buffer) else {
       return nil
     }
-    assert(CVPixelBufferGetPixelFormatType(buffer) == kCVPixelFormatType_32BGRA)
-    let count = CVPixelBufferGetDataSize(buffer)
-    let bufferData = Data(bytesNoCopy: mutableRawPointer, count: count, deallocator: .none)
-    var rgbBytes = [UInt8](repeating: 0, count: byteCount)
-    var pixelIndex = 0
-    for component in bufferData.enumerated() {
-      let bgraComponent = component.offset % bgraPixel.channels;
-      let isAlphaComponent = bgraComponent == bgraPixel.alphaComponent;
-      guard !isAlphaComponent else {
-        pixelIndex += 1
-        continue
-      }
-      // Swizzle BGR -> RGB.
-      let rgbIndex = pixelIndex * rgbPixelChannels + (bgraPixel.lastBgrComponent - bgraComponent)
-      rgbBytes[rgbIndex] = component.element
+    
+    let width = CVPixelBufferGetWidth(buffer)
+    let height = CVPixelBufferGetHeight(buffer)
+    let sourceBytesPerRow = CVPixelBufferGetBytesPerRow(buffer)
+    let destinationChannelCount = 3
+    let destinationBytesPerRow = destinationChannelCount * width
+    
+    var sourceBuffer = vImage_Buffer(data: sourceData,
+                                     height: vImagePixelCount(height),
+                                     width: vImagePixelCount(width),
+                                     rowBytes: sourceBytesPerRow)
+    
+    guard let destinationData = malloc(height * destinationBytesPerRow) else {
+      print("Error: out of memory")
+      return nil
+    }
+    
+    defer {
+      free(destinationData)
+    }
+
+    var destinationBuffer = vImage_Buffer(data: destinationData,
+                                          height: vImagePixelCount(height),
+                                          width: vImagePixelCount(width),
+                                          rowBytes: destinationBytesPerRow)
+    
+    if (CVPixelBufferGetPixelFormatType(buffer) == kCVPixelFormatType_32BGRA){
+      vImageConvert_BGRA8888toRGB888(&sourceBuffer, &destinationBuffer, UInt32(kvImageNoFlags))
+    } else if (CVPixelBufferGetPixelFormatType(buffer) == kCVPixelFormatType_32ARGB) {
+      vImageConvert_ARGB8888toRGB888(&sourceBuffer, &destinationBuffer, UInt32(kvImageNoFlags))
+    }
+
+    let byteData = Data(bytes: destinationBuffer.data, count: destinationBuffer.rowBytes * height)
+    if isModelQuantized {
+      return byteData
+    }
+
+    // Not quantized, convert to floats
+    let bytes = Array<UInt8>(unsafeData: byteData)!
+    var floats = [Float](repeating: 0, count: bytes.count)
```

Comment:
`count: bytes.count` should be changed to `count: 0`.
As you appends the value in for-loops, you don't need to initialize `bytes.count` sized array.

With changing `map` function to for-loop, I can find the time is reduced.
Thank you!

Reply:
If I remember correctly, I tried allocating the memory beforehand in an attempt to speed up the function, but it didn't have an effect. I'll change it to `[Float]()`.

Reply:
I'm still kind of surprised the append() here doesn't destroy performance, but if you've measured with an optimized build then that's fine.

Reply:
It was a while ago, so I don't remember how thorough my tests were on non-quantized data. However I think most variable arrays are allocated not a few bytes at a time but the allocation is doubled every time capacity is reached. That way the allocation complexity is is logN while putting stuff in the array is N.

Reply:
Right, but the allocation of memory can be a large constant in the complexity.  I'd feel better about using `reserveCapacity` or something like that.